resources/assets/js/components/CommentForm.tsx

```diff
@@ -0,0 +1,212 @@
+    defaultMessage: "eg. Enter your question, recommendation, etc.",
+    description: "The placeholder displayed for the comment input field.",
+  },
+  commentTypeLabel: {
+    id: "commentForm.commentType.label",
+    defaultMessage: "Type of Comment",
+    description: "The label displayed for the comment type select box.",
+  },
+  commentTypeNullSelection: {
+    id: "commentForm.commentType.nullSelection",
+    defaultMessage: "Select a comment type...",
+    description: "Comment type from list of comment types",
+  },
+});
+
+export const commentTypeMessages = defineMessages({
+  question: {
+    id: "commentType.question",
+    defaultMessage: "Question",
+    description: "Comment type from list of comment types",
+  },
+  recommendation: {
+    id: "commentType.recommendation",
+    defaultMessage: "Recommendation",
+    description: "Comment type from list of comment types",
+  },
+  requiredAction: {
+    id: "commentType.requiredAction",
+    defaultMessage: "Required Action",
+    description: "Comment type from list of comment types",
+  },
+  comment: {
+    id: "commentType.comment",
+    defaultMessage: "Comment",
+    description: "Comment type from list of comment types",
+  },
+});
+
+interface CommentFormProps {
+  jobId: number;
+  isHrAdviser: boolean;
+  location: string;
+  handleCreateComment: (jobId: number, newComment: Comment) => Promise<Comment>;
+}
+
+interface CommentFormValues {
+  comment: string;
+  commentType: number | null;
+}
+
+const CommentForm: React.FunctionComponent<CommentFormProps> = ({
+  isHrAdviser,
+  handleCreateComment,
+  jobId,
+  location,
+}: CommentFormProps): React.ReactElement => {
+  const intl = useIntl();
+  const initialValues: CommentFormValues = {
+    comment: "",
+    commentType: 0 || null,
+  };
+
+  const hrCommentSchema = Yup.object().shape({
+    comment: Yup.string().required(
+      intl.formatMessage(validationMessages.required),
+    ),
+    commentType: Yup.number()
+      .oneOf(
+        Object.values(CommentTypeId),
+        intl.formatMessage(validationMessages.invalidSelection),
+      )
+      .required(intl.formatMessage(validationMessages.required)),
+  });
+
+  const managerCommentSchema = Yup.object().shape({
+    comment: Yup.string().required(
+      intl.formatMessage(validationMessages.required),
+    ),
+  });
+
+  return (
+    <section>
+      <Formik
+        initialValues={initialValues}
+        validationSchema={isHrAdviser ? hrCommentSchema : managerCommentSchema}
+        onSubmit={(values, { setSubmitting, resetForm }): void => {
+          const newComment: Comment = {
+            ...emptyComment(),
+            location,
+            comment: values.comment,
+            type_id: isHrAdviser ? Number(values.commentType) : null,
+          };
+          handleCreateComment(jobId, newComment).finally((): void => {
+            setSubmitting(false); // Required by Formik to finish the submission cycle
```

Comment:
If the promise comes back with an error, we'll want to setSubmitting(false) but probably not reset the form.